### ISSUE TITLE
Support grids acting as layout containers

### ISSUE THREAD

**Issue by Andarist** [adobe/react-spectrum#1782]
# 🙋 Feature Request

Looking at the implementation of the [`useGrid`](https://github.com/adobe/react-spectrum/blob/887fff79e8f33c0f8953dc38955419feddd56b6a/packages/%40react-aria/grid/src/useGrid.ts) it seems that you are currently focusing on collection grids (as in tabular data) but [the Aria Practices document mentions 2 types of grids](https://github.com/adobe/react-spectrum/blob/887fff79e8f33c0f8953dc38955419feddd56b6a/packages/%40react-aria/grid/src/useGrid.ts): interactive tabular data and layout containers.

I believe that the latter case should usually not be searchable nor selectable (and maybe some other behaviors should be turned off for it as well). Both of those seem to be enabled by default in the `useGrid` and there is no option to disable those.

**Comment by devongovett** [comment]
Yes, we plan to use the grid implementation for things like a tag list which is one of the examples in aria practices - https://www.w3.org/TR/wai-aria-practices-1.2/examples/grid/LayoutGrids.html

As for selection, it can be turned off using `selectionMode="none"`, and we could probably add a prop to disable type select as well.
